### PR TITLE
Gets the legend images in parallel

### PR DESCRIPTION
**Pull request by sbrunner**
Fix [GSGMF-2081](https://jira.camptocamp.com/browse/GSGMF-2081)

### CODE REVIEW

File: tilecloud_chain/controller.py [hunk 1]
```diff
@@ -279,46 +310,64 @@ def _fill_legend(
...
+    for future in concurrent.futures.as_completed(legend_image_future):
+        layer_name, resolution = legend_image_future[future]
+        try:
+            legend_image_metadata.setdefault(layer_name, {})[resolution] = future.result()
+        except Exception as exc:
+            _LOGGER.warning(
+                "Unable to get legend image for layer '%s', resolution '%s': %s", layer_name, resolution, exc
+            )
+
+    _LOGGER.info("Get %i legend images in %s", len(legend_image_future), time.perf_counter() - start)
```
Comment (danduk82) [inline tilecloud_chain/controller.py:353]: does this need to be in `INFO` log level? seems quite verbose in a production env. It should be more `DEBUG` probably

File: tilecloud_chain/controller.py [hunk 2]
```diff
@@ -279,46 +310,64 @@ def _fill_legend(
         assert gene.config_file
         config = gene.get_config(gene.config_file)
 
+    start = time.perf_counter()
+    legend_image_future = {}
+    with concurrent.futures.ThreadPoolExecutor(
+        max_workers=int(os.environ.get("TILECLOUD_CHAIN_CONCURENT_GET_LEGEND", "20"))
```
Comment (danduk82) [inline tilecloud_chain/controller.py:316]: please document in the readme the usage of `TILECLOUD_CHAIN_CONCURENT_GET_LEGEND`

**Comment by sbrunner** [comment]
@danduk82 Done :-)

File: tilecloud_chain/USAGE.rst [hunk 3]
```diff
@@ -755,6 +755,8 @@ Environment variables
 - ``TILE_CHUNK_SIZE``: Default to ``1``, the chunk size to process the tiles after the meta tiles.
 - ``TILECLOUD_CHAIN_MAX_OUTPUT_LENGTH``: Default to ``1000``, the maximum number of character of the
   output to be display in the admin interface.
+- ``TILECLOUD_CHAIN_CONCURENT_GET_LEGEND``: Default to ``10``, the number of concurrent request to get the
```
Comment (danduk82) [inline tilecloud_chain/USAGE.rst:758]: default in the code is `20` please decide if you put `20` here or `10` in the code ;P

**Review by danduk82 — CHANGES_REQUESTED** [review]

File: tilecloud_chain/USAGE.rst (hunk 3 above)
Comment (sbrunner) [reply tilecloud_chain/USAGE.rst:758]: => 10 everywhere :-)

**Review by danduk82 — APPROVED** [review]

**Pull request merged**